cmd/apiserver/server_test.go

```diff
@@ -0,0 +1,462 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	calculationsv1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	v1 "github.com/vega-project/ccb-operator/pkg/apis/calculations/v1"
+	"github.com/vega-project/ccb-operator/pkg/client/clientset/versioned/fake"
+	"github.com/vega-project/ccb-operator/pkg/util"
+)
+
+func TestCreateCalculation(t *testing.T) {
+	testCases := []struct {
+		id                 string
+		jsonStr            []byte
+		initialCalculation calculationsv1.Calculation
+		expected           calculationsv1.Calculation
+	}{
+		{
+			id: "one calculation gets created",
+			initialCalculation: calculationsv1.Calculation{
+				TypeMeta:   metav1.TypeMeta{Kind: "Calculation", APIVersion: "vega.io/v1"},
+				ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"created_by_human": "true"}, Name: fmt.Sprintf("calc-%s", util.InputHash([]byte(fmt.Sprintf("%f", 12100.0)), []byte(fmt.Sprintf("%f", 4.0))))},
+				Phase:      calculationsv1.CreatedPhase,
+				Status:     calculationsv1.CalculationStatus{StartTime: metav1.Time{Time: time.Now()}},
+				Spec: calculationsv1.CalculationSpec{Teff: 12100.0, LogG: 4.0, Steps: []v1.Step{
+					{
+						Command: "atlas12_ada",
+						Args:    []string{"s"},
+					},
+					{
+						Command: "atlas12_ada",
+						Args:    []string{"r"},
+					},
+					{
+						Command: "synspec49",
+						Args:    []string{"<", "input_tlusty_fortfive"},
+					},
+				}},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		fakecs := fake.NewSimpleClientset()
+		fakeClient := fakecs.VegaV1()
+
+		var calc struct {
+			Teff string
+			LogG string
+		}
+
+		s := server{
+			logger: logrus.WithField("test-name", "create calculations test"),
+			ctx:    context.Background(),
+			client: fakeClient,
+		}
+
+		tc.expected = tc.initialCalculation
```

Comment:
lets don't do this. Please define the expected in the testcase struct as well like the `initialCalculations`